pkg/controller/dns/dns_policy.go

```diff
@@ -0,0 +1,204 @@
+	gcp "github.com/crossplane-contrib/provider-gcp/pkg/clients"
+	dnsclient "github.com/crossplane-contrib/provider-gcp/pkg/clients/dns"
+	"github.com/crossplane-contrib/provider-gcp/pkg/features"
+)
+
+const (
+	errorGetFailed    = "cannot get the DNSPolicy"
+	errorNotDNSPolicy = "managed resource is not a DNSPolicy custom resource"
+	errorCannotDelete = "cannot delete new DNSPolicy"
+	errorCreatePolicy = "cannot create DNSPolicy"
+	// errorManagedUpdateFailed = "cannot update DNSPolicy custom resource"
+	// err_CannotCreate        = "cannot create new DNSPolicy"
+	// errorCheckUpToDate       = "cannot determine if DNSPolicy is up to date"
+)
+
+// SetupPolicy adds a controller that reconciles
+// DNSPolicy managed resources.
+func SetupPolicy(mgr ctrl.Manager, o controller.Options) error {
+	name := managed.ControllerName(v1alpha1.PolicyGroupKind)
+
+	cps := []managed.ConnectionPublisher{managed.NewAPISecretPublisher(mgr.GetClient(), mgr.GetScheme())}
+	if o.Features.Enabled(features.EnableAlphaExternalSecretStores) {
+		cps = append(cps, connection.NewDetailsManager(mgr.GetClient(), scv1alpha1.StoreConfigGroupVersionKind))
+	}
+
+	r := managed.NewReconciler(mgr,
+		resource.ManagedKind(v1alpha1.PolicyGroupVersionKind),
+		managed.WithExternalConnecter(&Connector{kube: mgr.GetClient()}),
+		managed.WithInitializers(managed.NewNameAsExternalName(mgr.GetClient())),
+		managed.WithPollInterval(o.PollInterval),
+		managed.WithLogger(o.Logger.WithValues("controller", name)),
+		managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name))),
+		managed.WithConnectionPublishers(cps...))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		WithOptions(o.ForControllerRuntime()).
+		For(&v1alpha1.Policy{}).
+		Complete(ratelimiter.NewReconciler(name, r, o.GlobalRateLimiter))
+}
+
+// Connector is ..
+type Connector struct {
+	kube client.Client
+}
+
+// Connect is
+func (c *Connector) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	projectID, opts, err := gcp.GetConnectionInfo(ctx, c.kube, mg)
+	if err != nil {
+		return nil, err
+	}
+
+	d, err := dns.NewService(ctx, opts...)
+	if err != nil {
+		return nil, errors.Wrap(err, errNewClient)
+	}
+	return &policyExternal{
+		kube:      c.kube,
+		dns:       d.Policies,
+		projectID: projectID,
+	}, nil
+}
+
+type policyExternal struct {
+	kube      client.Client
+	dns       *dns.PoliciesService
+	projectID string
+}
+
+func (e *policyExternal) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mg.(*v1alpha1.Policy)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errorNotDNSPolicy)
+	}
+
+	policy, err := e.dns.Get(
+		e.projectID,
+		meta.GetExternalName(cr),
+	).Context(ctx).Do()
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(
+			resource.Ignore(gcp.IsErrorNotFound, err),
+			errorGetFailed,
+		)
+	}
+
+	cr.SetConditions(xpv1.Available())
+
```

Comment:
Seems like `ID` is the only field in status. We should set it here with the information retrieved with `Get` call above.